packages/lib/src/components/GooglePay/GooglePay.tsx

```diff
@@ -43,8 +43,8 @@ class GooglePay extends UIElement<GooglePayProps> {
     public loadPayment = () => {
         const { onSubmit = () => {}, onAuthorized = () => {} } = this.props;
 
-        return this.googlePay
-            .initiatePayment(this.props)
+        return new Promise((resolve, reject) => (this.props.onClick ? this.props?.onClick(resolve, reject) : resolve()))
+            .then(() => this.googlePay.initiatePayment(this.props))
```

Comment:
Another possibility would be to add the `onClick` to the default props making it just call the resolve function. Then can just call it without checking if it's defined or not.

Reply:
👍